server/src/main/java/io/crate/planner/node/ddl/AlterTableAddColumnPlan.java

```diff
@@ -100,13 +102,17 @@ public void executeOrFail(DependencyCarrier dependencies,
                 tableElements
             );
 
-            List<Reference> references = new ArrayList<>();
+            // We can add multiple object columns via ALTER TABLE ADD COLUMN.
+            // Those columns can have overlapping paths, for example we can add columns o['a']['b'] and o['a']['c'].
+            // For every added column AnalyzedColumnDefinition provides not only leaf but also path to the root.
+            // For o['a']['b'] and o['a']['c'] we can end up having Ref(Ident(o)) and Ref(Ident(a)) twice.
+            Set<Reference> references = new HashSet<>();
             IntArrayList pKeysIndices = new IntArrayList();
```

Comment:
`references` needs to keep insertion order, otherwise the `pKeysIndices` won't work as they're referring to specific positions.

Reply:
Good catch, thanks. Added a unit test to cover this case. Also, even cutting duplicates was done wrongly, need to rely on `FQN` but not on `Reference.equals`, see https://github.com/crate/crate/pull/13383/commits/9c2b8a17a8b57c47dc77d83914c46314417db155#diff-b3e9bec0a4a8f540cf3dec4580badee7db16c5093337574518dd3169c10d3647R138-R140